pkg/kube/pod_runner_test.go

```diff
@@ -109,9 +114,84 @@ func (s *PodRunnerTestSuite) TestPodRunnerForSuccessCase(c *C) {
 	cancel()
 }
 
+// TestPodRunnerWithDebugLabelForSuccessCase adds a debug entry into the context and verifies the
+// pod got created with corresponding label using the entry or not.
+func (s *PodRunnerTestSuite) TestPodRunnerWithDebugLabelForSuccessCase(c *C) {
+	jobIDSuffix := "JobID"
+	for _, tc := range []struct {
+		name          string
+		targetKey     string
+		contextKey    string
+		contextValue  string
+		expectedLabel bool
+	}{
+		{
+			name:          "target key (kanister.io/JobID) present in pod labels",
+			targetKey:     path.Join(consts.LabelPrefix, jobIDSuffix),
+			contextKey:    path.Join(consts.LabelPrefix, jobIDSuffix),
+			contextValue:  "xyz123",
+			expectedLabel: true,
+		},
+		{
+			name:          "target key (kanister.io/JobID) not present in pod labels",
+			targetKey:     path.Join(consts.LabelPrefix, jobIDSuffix),
+			contextKey:    path.Join(consts.LabelPrefix, "NonJobID"),
+			contextValue:  "some-other-value",
+			expectedLabel: false,
+		},
+	} {
+		ctx, cancel := context.WithCancel(context.Background())
+		ctx = field.Context(ctx, tc.contextKey, tc.contextValue)
+		cli := fake.NewSimpleClientset()
+		cli.PrependReactor("create", "pods", func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+			return false, nil, nil
+		})
+		cli.PrependReactor("get", "pods", func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+			p := &corev1.Pod{
+				Status: corev1.PodStatus{
+					Phase: corev1.PodRunning,
+				},
+			}
+			return true, p, nil
+		})
+		po := &PodOptions{
+			Namespace: podRunnerNS,
+			Name:      podName,
+			Command:   []string{"sh", "-c", "tail -f /dev/null"},
+		}
+		deleted := make(chan struct{})
+		cli.PrependReactor("delete", "pods", func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+			c.Log("Pod deleted due to Context Cancelled")
+			close(deleted)
+			return true, nil, nil
+		})
+		AddLabelsToPodOptionsFromContext(ctx, po, tc.targetKey)
+		pr := NewPodRunner(cli, po)
+		errorCh := make(chan error)
+		go func() {
+			_, err := pr.Run(ctx, afterPodRunTestKeyPresentFunc(tc.targetKey, tc.expectedLabel, deleted))
```

Comment:
We can pass expected label value and validate if we are receiving the same 
```suggestion
			_, err := pr.Run(ctx, afterPodRunTestKeyPresentFunc(tc.targetKey, tc.isLabelExpected, tc.targetLabel, deleted))
```

Reply:
Done